contracts/PositionManager.sol

```diff
@@ -96,17 +96,39 @@ contract PositionManager is IPositionManager, ERC721Holder {
         address _uniswapAddressHolder,
         address _registry,
         address _aaveAddressHolder
-    ) public {
+    ) external {
         StorageStruct storage Storage = PositionManagerStorage.getStorage();
         Storage.owner = _owner;
         Storage.uniswapAddressHolder = IUniswapAddressHolder(_uniswapAddressHolder);
         Storage.registry = IRegistry(_registry);
         Storage.aaveAddressHolder = IAaveAddressHolder(_aaveAddressHolder);
     }
 
+    ///@notice withdraw uniswap position NFT from the position manager
+    ///@param tokenId ID of withdrawn token
+    function withdrawUniNft(uint256 tokenId) external override onlyOwner {
+        StorageStruct storage Storage = PositionManagerStorage.getStorage();
+
+        uint256 index = uniswapNFTs.length;
+        for (uint256 i = 0; i < uniswapNFTs.length; i++) {
+            if (uniswapNFTs[i] == tokenId) {
+                index = i;
+                i = uniswapNFTs.length;
+            }
+        }
+        require(index < uniswapNFTs.length, 'PositionManager::withdrawUniNFT: token ID not found!');
+        INonfungiblePositionManager(Storage.uniswapAddressHolder.nonfungiblePositionManagerAddress()).safeTransferFrom(
+            address(this),
+            msg.sender,
+            tokenId,
+            '0x0'
+        );
+        emit WithdrawUni(msg.sender, tokenId);
+    }
+
     ///@notice remove awareness of tokenId UniswapV3 NFT
     ///@param tokenId ID of the NFT to remove
-    function removePositionId(uint256 tokenId) public override {
+    function removePositionId(uint256 tokenId) external override {
```

Comment:
Does this change keep tests running? Because when we use delegate call, position manager may call this function

Reply:
Not for now, so better keep that external for now